wormhole-connect/src/utils/routes/cctpRelay.ts

```diff
@@ -0,0 +1,696 @@
+            title: 'Relayer fee',
+            value: relayerFee ? `${relayerFee} ${token.symbol}` : '—',
+          },
+        ],
+      },
+    ];
+  }
+
+  public getNativeBalance(
+    address: string,
+    network: ChainName | ChainId,
+  ): Promise<BigNumber | null> {
+    return wh.getNativeBalance(address, network);
+  }
+
+  public getTokenBalance(
+    address: string,
+    tokenId: TokenId,
+    network: ChainName | ChainId,
+  ): Promise<BigNumber | null> {
+    return wh.getTokenBalance(address, tokenId, network);
+  }
+
+  async getRelayerFee(
+    sourceChain: ChainName | ChainId,
+    destChain: ChainName | ChainId,
+    token: string,
+  ): Promise<BigNumber> {
+    const tokenConfig = TOKENS[token];
+    if (!tokenConfig) throw new Error('could not get token config');
+    const tokenId = tokenConfig.tokenId;
+
+    // only works on EVM
+    const chainContext = wh.getContext(
+      sourceChain,
+    ) as EthContext<WormholeContext>;
+    const circleRelayer =
+      chainContext.contracts.mustGetWormholeCircleRelayer(sourceChain);
+    const destChainId = wh.toChainId(destChain);
+    const fee = await circleRelayer.relayerFee(destChainId, tokenId?.address);
+    return fee;
+  }
+
+  async getForeignAsset(
+    token: TokenId,
+    chain: ChainName | ChainId,
+  ): Promise<string | null> {
+    // assumes USDC
+    const addr = TOKENS_ARR.find(
+      (t) => t.symbol === 'USDC' && t.nativeNetwork === chain,
+    )?.tokenId?.address;
+    if (!addr) throw new Error('USDC not found');
+    return addr;
+  }
+
+  async getMessageInfo(
+    tx: string,
+    chain: ChainName | ChainId,
+  ): Promise<CCTPInfo> {
+    // only EVM
+    // use this as reference
+    // https://goerli.etherscan.io/tx/0xe4984775c76b8fe7c2b09cd56fb26830f6e5c5c6b540eb97d37d41f47f33faca#eventlog
+    const provider = wh.mustGetProvider(chain);
+
+    const receipt = await provider.getTransactionReceipt(tx);
+    if (!receipt) throw new Error(`No receipt for ${tx} on ${chain}`);
+
+    const vaaInfo: VaaInfo = await wh.getVaa(tx, chain);
+    const payload = vaaInfo.vaa.payload.toString('hex');
+
+    // Get the CCTP log
+    const cctpLog = receipt.logs.filter(
+      (log) =>
+        log.topics[0] ===
+        '0x2fa9ca894982930190727e75500a97d8dc500233a5065e0f3126c48fbe0343c0',
+    )[0];
+
+    const parsedCCTPLog = new utils.Interface([
+      'event DepositForBurn(uint64 indexed nonce, address indexed burnToken, uint256 amount, address indexed depositor, bytes32 mintRecipient, uint32 destinationDomain, bytes32 destinationTokenMessenger, bytes32 destinationCaller)',
+    ]).parseLog(cctpLog);
+
+    const messageLog = receipt.logs.filter(
+      (log) =>
+        log.topics[0] ===
+        '0x8c5261668696ce22758910d05bab8f186d6eb247ceac2af2e82c7dc17669b036',
+    )[0];
+
+    const message = new utils.Interface([
+      'event MessageSent(bytes message)',
+    ]).parseLog(messageLog).args.message;
+
+    const messageHash = utils.keccak256(message);
```

Comment:
does this work? 

Locally I had to convert the message to bytes _then_ hash it _then_ apply `0x` prefix to get the actual Attestation from circle



```ts
    const [messageLog] = messageLogs;
    // Need to get the message (0xdeadbeef...) to bytes prior to hashing
    const msgBytes = new Uint8Array(
      Buffer.from((messageLog.args.message as string).slice(2), 'hex'),
    );
    const messageHash = `0x${Buffer.from(keccak256(msgBytes)).toString('hex')}`;
```